src/Shared/Hooks/useForm/useForm.ts

```diff
@@ -81,6 +103,12 @@ export const useForm = <T extends Record<keyof T, any> = {}>(options?: {
      */
     const handleSubmit = (onValid: UseFormSubmitHandler<T>) => (e: FormEvent<HTMLFormElement>) => {
         e.preventDefault()
+
+        // Enables validation for all form fields if not enabled after form submission.
+        if (Object.keys(enableValidationOnChange) !== Object.keys(data)) {
```

Comment:
FYI, we cannot compare arrays

Reply:
this was supposed to be length missed it